server/routes/api/place.py

```diff
@@ -130,6 +130,67 @@ def api_name():
     return Response(json.dumps(result), 200, mimetype='application/json')
 
 
+@cache.memoize(timeout=3600 * 24)  # Cache for one day.
+def cached_i18n_name(dcids, locale):
+    """Returns localination names for set of dcids.
+
+    Args:
+        dcids: ^ separated string of dcids. It must be a single string for the cache.
+        locale: the desired localization language code.
+
+    Returns:
+        A dictionary of place international names, keyed by dcid.
+    """
+    dcids = dcids.split('^')
+    response = fetch_data('/node/property-values', {
+        'dcids': dcids,
+        'property': 'nameWithLanguage',
+        'direction': 'out'
+    },
+                          compress=False,
+                          post=True)
+    result = {}
+    for dcid in dcids:
+        values = response[dcid].get('out')
+        name_in_en = ''
+        result[dcid] = ''
+        for entry in values:
+            if entry['value'].endswith(locale):
```

Comment:
should we test for "@" as well? e.g. endswith("@en")
it would be good to also lowercase the locale

also, not sure if we have expanded locale info in the graph (e.g. pt-br). we might get those expanded codes though, so would be good to take the appropriate part of the locale.

that said, we can simplify this function and only accept simplified locale info, and do the user locale parsing elsewhere.

Reply:
Done lowercase. 

Regarding testing with endswith("@en"), shall the url be like hl=en instead of hl=@en? 
If yes, don't think endswith("@en") is needed then. 

We have pt-br as well as pt, and we also have both zh-cn & zh. 
So don't think we need to tweak the parameter here except lower case?

Reply:
my thought with adding the "@" sign in the tests are to capture cases such as:
hl=ru
i18nname=foobar@ru | foobar@ru-RU | foobar@tt-ru

(we don't have those languages, but there might be a collision between locale and language?)

btw, are we storing locale as uppercase or lowercase? we might need to lowercase the returned value too.

(yes, it should be hl=en not hl=@en)

Reply:
Ic. Tests added. 
We store all the locale as lower case. So convert the parameter to lower case shall be good.

Reply:
I added both fallback case, from it-eu fall back to it then fall back to en. 
Also updated the test to capture @en-ru when we want @ru, nice catch!
(I have to use ends with but not finds otherwise @en-ru would be return when we want @en.